alibi_detect/utils/_types.py

```diff
@@ -17,12 +17,12 @@
 supported_optimizers_tf, supported_optimizers_torch = None, None
 if has_tensorflow:
     import tensorflow as tf
-    supported_models_tf = tf.keras.Model
+    supported_models_tf = (tf.keras.Model, )
     supported_optimizers_tf = (tf.keras.optimizers.Optimizer, type)
 if has_pytorch:
     import torch
-    supported_models_torch = torch.nn.Module
-    supported_optimizers_torch = type  # Note type not object!
-supported_models_sklearn = BaseEstimator
+    supported_models_torch = (torch.nn.Module, )
+    supported_optimizers_torch = (type, )  # Note type not object!
+supported_models_sklearn = (BaseEstimator, )
 supported_models_all = (supported_models_tf, supported_models_torch, supported_models_sklearn)
 supported_optimizers_all = (supported_optimizers_tf, supported_optimizers_torch)
```

Comment:
Is this a bug now that the members are tuples of length 1 (if so, why isn't anything failing?)? Should we use addition to create a new tuple?

Reply:
Passing tuples of tuples to `isinstance` seems to be fine. For example:

```python
a = 2
test = ((str,), (float,))
isinstance(a, test)  # gives False
```

However, I agree its ugly, so will change to `supported_optimizers_tf + supported_optimizers_torch`